Linter: Implement `erb-no-output-control-flow` rule

written-fresh opened the pull request:
ERB no output control flow disallows rendering if/else/elsif/end blocks

In the future we could expand this rule to also catch `for`/`while`/`until` and maybe more.

Closes https://github.com/marcoroth/herb/issues/126

written-fresh committed:
feat(linter): Add rule for not outputting from control flow tags

- erb no output control flow disallows rendering if/else/elsif/end blocks

marcoroth reviewed (commented):
Hey @drewhoffer, thanks for this pull request! 

Just a first pass-through that I saw when looking over it 🙌🏼

written-fresh committed:
add recursion

- recursively check nodes inside statements
- fix incorrectly named test file
- write tests
- reorder imports

written-fresh committed:
apply code suggestions

marcoroth commented on javascript/packages/linter/src/rules/erb-no-output-control-flow.ts:
```diff
@@ -0,0 +1,66 @@
+    this.visitChildNodes(node)
+  }
+  
+  getMessages(): LintMessage[] {
+    return this.messages
+  }
```
I think we can just remove this one, we removed the interface for this in #203

marcoroth commented on javascript/packages/linter/src/rules/erb-no-output-control-flow.ts:
```diff
@@ -0,0 +1,66 @@
+
+export class ERBNoOutputControlFlow implements Rule {
+  name = "erb-no-output-control-flow"
+  description = "Prevent block-level elements from being placed inside inline elements"
```
we also removed this description in #203

written-fresh committed:
update tests

written-fresh committed:
remove unused description

written-fresh committed:
fix test html output

written-fresh committed:
fix md doc

written-fresh committed:
remove getMessages call

written-fresh committed:
refactor to use BaseRuleVisitor

written-fresh committed:
remove unused messages

marcoroth reviewed (approved):
Awesome, thank you so much @drewhoffer! 🙌🏼
